[Mant Koh En Wei] iP

M-kew opened the pull request:
## Nuffle

> “SUIIIIIIIIIII” – Cristiano Ronaldo

**Nuffle** is here to make your workflow smoother, simpler, and more efficient. With it, you can:

- Manage tasks effortlessly
- Keep track of your deadlines
- Organize your notes

It's **easy to learn** and **blazing fast** to use. 🚀

### How to Get Started:

1. [Download the latest release](https://example.com).
2. Run the `.jar` file.
3. Start organizing your life!

And did I mention? It's **FREE**! 🎉

### Features:

- [x] Task management
- [ ] Deadline reminders (coming soon)
- [ ] Priority tags (coming soon)

If you're a Java enthusiast, you'll love tinkering with the code. Here's a sneak peek at the `Main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

M-kew committed:
Implemented skeletal version of chatbot.

M-kew committed:
Implemented echo feature for Nuffle chatbot.

M-kew committed:
Implemented list feature to the chatbot.

M-kew committed:
Implemented mark as done feature on chatbot.

M-kew committed:
Implemented three type of tasks (todo, event, deadline) to the chatbot.

M-kew committed:
Implemented text UI testing for chatbot.

M-kew committed:
Implemented exception handling in Nuffle chatbot.

M-kew committed:
Implemented exception handling in Nuffle chatbot.

M-kew committed:
Implemented delete feature in Nuffle chatbot.

M-kew committed:
Implemented loading of file into the chatbot at startup and saving file

M-kew committed:
Merge branch 'branch-Level-7'

M-kew committed:
Implemented date and time parser for Nuffle chatbot

M-kew committed:
Implement OOP by adding Parser, Storage, Ui and TaskList

M-kew committed:
Implement OOP by adding Storage, Ui, TaskList and Parser

M-kew committed:
Merge branch level 8 into master

M-kew committed:
Merge branch MoreOOP into master

M-kew committed:
Package java classes used in Nuffle chatbot

M-kew committed:
Merge branch A-Packages into master

M-kew committed:
Merge remote-tracking branch 'origin/add-gradle-support'

M-kew committed:
Create Junit test cases for TaskList and Parser. Implement gradle to Nuffle chatbot project

M-kew committed:
Merge branch A-Junit into master

M-kew committed:
Implement find function in Nuffle chatbot

M-kew committed:
Add javadocs to classes and methods

M-kew committed:
Apply java coding standard to scripts

M-kew committed:
Merge tag 'A-CodingStandard'

M-kew committed:
Merge branch 'branch-Level-9'

M-kew committed:
modify Nuffle.java methods signature

M-kew committed:
Implement GUI to nufflebot

Fluffykan commented on src/test/java/nuffle/TaskListTest.java:
```diff
@@ -0,0 +1,22 @@
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class TaskListTest {
```
I like how you have created test cases for testing the feature which adds tasks to the task list! Just remember to add on to the testcases as you implement more functionalities to your `TaskList` class

Fluffykan commented on src/main/java/nuffle/ui/Ui.java:
```diff
@@ -0,0 +1,141 @@
+     *
+     * @param task The task that has been marked as not done.
+     */
+    public static void unmarkTaskMessage(Task task) {
```
perhaps these print statements could be placed under the `TaskList` class, as it is just a print statement and might not require its own method

Fluffykan commented on src/test/java/nuffle/parserTest.java:
```diff
@@ -0,0 +1,29 @@
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class parserTest {
```
perhaps you could create tests for strings that are not formatted properly, e.g. missing deadline, missing spaces between some words, etc. to see if your `Parser` class can correctly catch such errors and either throw a checked exception or reject the user's input, instead of potentially crashing the entire program

Fluffykan commented on src/main/java/nuffle/ui/Ui.java:
```diff
@@ -0,0 +1,141 @@
+    /**
+     * Displays the welcome message when the application starts.
+     */
+    public static void welcomeMessage() {
```
perhaps the method's name could be `printWelcomeMessage` instead, which is phrased to sound more like an action than a noun

Fluffykan commented on src/main/java/nuffle/task/Deadline.java:
```diff
@@ -0,0 +1,71 @@
+     *
+     * @return A string representation of the deadline task suitable for saving to a file.
+     */
+    public String saveFormat() {
```
perhaps this method could be renamed as `printSaveFormat` to make it sound like a verb instead of a noun

Fluffykan commented on src/main/java/nuffle/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+            String[] components = eachLine.split("\\|");
+
+            String category = components[0].trim();
+            String desc = components[2].trim();
```
perhaps you might want to consider checking the size of the array before doing any operations on it in case of an IndexOutOfBoundsException should the save file be corrupted, e.g. missing a field like D | return book | June 6th

M-kew committed:
Implement changes based on pull request reviews for NuffleBot

M-kew committed:
Include assertion in code logic for Nuffle.java and Parser.java

M-kew committed:
Implement new class Command to ensure code quality in Main.java

M-kew committed:
Implement new class Command to ensure code quality in Main.java

M-kew committed:
The backend logic in `Nuffle.java` for handling user commands is too long and tightly coupled, making it difficult to manage and extend.

This code structure violates the single responsibility principle, leading to decreased maintainability, and making future enhancements more complex.

Extract the command handling logic into separate `Command` classes for each specific action (e.g., `MarkCommand`, `DeadlineCommand`). This decouples the responsibilities and modularizes the code.

Creating individual classes for commands allows for better separation of concerns, easier testing, and more scalable code that can be extended without cluttering `Nuffle.java`.

These changes pave the way for future improvements, making the code more robust and easier to understand for new developers.

M-kew committed:
Implement loan feature into Nufflebot.

M-kew committed:
Merge pull request #1 from M-kew/branch-D-Loans

BCD-Extensions (D-Loans)

M-kew committed:
Modify loan command to handle incorrect format of due date

M-kew committed:
Tweak GUI elements to make chatbot visually pleasing

M-kew committed:
Tweak GUI elements to make chatbot visually pleasing

M-kew committed:
Tweak chatbot interactive text to be more personalised

M-kew committed:
Implement Junit testing for find and delete command

M-kew committed:
Tweak GUI to include chatbot name in window title

M-kew committed:
Implement file error handling in nuffle chatbot

M-kew committed:
Add README user guide for Nuffle chatbot

M-kew committed:
Modify README user guide
